pandas/tests/indexing/test_interval.py

```diff
@@ -122,6 +148,46 @@ def test_with_slices(self):
         with pytest.raises(ValueError):
             s[0:4:2]
 
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_with_slices_updated_behavior(self):
+
+        s = self.s
+
+        # slice of interval
+        expected = s.iloc[4:]
+        result = s.loc[Interval(3, 4):]
+        tm.assert_series_equal(expected, result)
+
+        expected = s.iloc[4:]
+        result = s[Interval(3, 4):]
+        tm.assert_series_equal(expected, result)
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 6):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 6):]
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 4, closed='left'):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 4, closed='left'):]
+
+        with pytest.raises(KeyError):
+            s.loc[Interval(3, 4, closed='both'):]
+
+        with pytest.raises(KeyError):
+            s[Interval(3, 4, closed='both'):]
+
+        # slice of scalar
+        with pytest.raises(NotImplementedError):
+            s[0:4] ## not sure what the behvaior should be here.
```

Comment:
This might indeed be a bad idea .. :-) But I think the following is probably the most consistent and safe given the current indexing mess/complexity:

- `df[:2]` does positional indexing (so similar to `df.iloc[:2]` as it does for all other types of indexing (apart from float index, which I think we should regard as a bug)
- disallow `df[:2.0]` ? If you want label based slicing based on the interval ranges, you can use `df.loc[:2.0]` (although we still have to decide what the semantics of that should be)

Apart from the integers (first case), also slicing with exact Interval objects would be allowed.